src/main/java/com/kulik/airbnb/service/AuthService.java

```diff
@@ -31,25 +33,26 @@ public String authenticate(AuthRequest request) {
         try {
             authenticationManager.authenticate(
                     new UsernamePasswordAuthenticationToken(request.getEmail(), request.getPassword()));
-            String token = jwtTokenProvider.createToken(request.getEmail());
+
+            if (userDao.getConfirmationFlag(request.getEmail()) == 0) {
+                throw new Exception("Account is not confirmed");
+            }
+
+            String token = jwtTokenProvider.createAuthorizationToken(request.getEmail());
```

Comment:
consider using a custom authentication manager https://stackoverflow.com/questions/31826233/custom-authentication-manager-with-spring-security-and-java-configuration